FRW/Services/Manager/EVMAccountManager.swift

```diff
@@ -167,7 +167,7 @@ extension EVMAccountManager {
             return
         }
 
-        if let EVMAddress = cacheAccounts[primaryAddress]?.first {
+        if let EVMAddress = cacheAccounts[primaryAddress]?.first, EVMAddress.count > 2 {
```

Comment:
Use `if let evmAddress = EthereumAddress(EVMAddress)` instead of `EVMAddress.count > 2`